rotkehlchen/api/rest.py

```diff
@@ -3945,6 +3945,53 @@ def import_user_assets(self, path: Path) -> Response:
 
         return api_response(OK_RESULT, status_code=HTTPStatus.OK)
 
+    def get_user_db_snapshot(self, timestamp: Timestamp) -> Response:
+        dbsnapshot = DBSnapshot(
+            db_handler=self.rotkehlchen.data.db,
+            msg_aggregator=self.rotkehlchen.msg_aggregator,
+        )
+        balances = dbsnapshot.get_timed_balances(timestamp)
+        location_data = dbsnapshot.get_timed_location_data(timestamp)
+        if len(balances) == 0 or len(location_data) == 0:
```

Comment:
It's never a good thing to assume that a third-party input is clean

Reply:
I love you both and happy to see discussions like this being civil :hugs: 

Checking api data should be done yeah. We should have 0 assumptions for anything coming outside of our app. At some point there may be more api consumers than just the frontend,

Reply:
Haha @nebolax you assume I read the PR. I have not. I only commented on the comment of checking data coming from outside.

So now you force me to **actually spend more time readin the commented part of the code**!!!!111eoneoneon.

Anyway ... hmm .. so looking at it indeed the lists do come from the DB, but the timestamp itself comes from the outside, right?

So if the api consumer provides a timestamp for which there is no snapshot we need to inform them about it. We can't just continue and let it fail with an exception later on?

Reply:
Oh, sorre :)
> you assume I read the PR. I have not

From what I see there will be no error, just empty lists are returned in case if there is no data found in the db. And my comment was about that this `if` can be redundant. So the communication problem was only about that I didn't understand how `It's never a good thing to assume that a third-party input is clean` is related here.

Reply:
But is absolutely not a problem that deserves long discussion ;)

Reply:
It's better to return error if no data is found for this api endpoint. So long as it's also clear in the docs I think it's better to be explicit rather than implicit.

Reply:
Okay, cool :+1: